### ISSUE TITLE
Duplicated rows / Updates are not deleted in iceberg table

### ISSUE THREAD

**Issue by julien-alpaca** [tabular-io/iceberg-kafka-connect#318]
The sink is not deleting the previous key when receiving an update although I have those two options enabled:

```
    "iceberg.tables.cdc-field": "__op",
    "iceberg.tables.upsert-mode-enabled": "true",
```

I specified the key and partition like this:
```
...
    "iceberg.tables.default-id-columns": "id",
    "iceberg.tables.default-partition-by": "day(created_at)",
```

Looks like the sink connector won't delete records created by a debezium snapshot (__op = 'r').

**Comment by julien-alpaca** [comment]
@bryanck Any idea?

**Comment by patduin** [comment]
@julien-alpaca I've been investigating similar thing and I believe I have a fix as well: https://github.com/databricks/iceberg-kafka-connect/pull/332

**Comment by julien-alpaca** [comment]
> [@julien-alpaca](https://github.com/julien-alpaca) I've been investigating similar thing and I believe I have a fix as well: [#332](https://github.com/databricks/iceberg-kafka-connect/pull/332)

@patduin the issue was because the delete event in kafka only contains the id value and the partition field was null. The sink can only delete rows that are in the same partition as per [iceberg specs](https://iceberg.apache.org/spec/#scan-planning). In our case we set `REPLICA IDENTITY FULL` on the source postgres database so that Debezium add all columns to the deleted row and not only the id.

-> that's for the deletes

In case of the updates, the sink was sometimes timing out when comitting files to GCS and increasing the timeout fixed the issue.

**Comment by patduin** [comment]
> > [@julien-alpaca](https://github.com/julien-alpaca) I've been investigating similar thing and I believe I have a fix as well: [#332](https://github.com/databricks/iceberg-kafka-connect/pull/332)
> 
> [@patduin](https://github.com/patduin) the issue was because the delete event in kafka only contains the id value and the partition field was null. The sink can only delete rows that are in the same partition as per [iceberg specs](https://iceberg.apache.org/spec/#scan-planning). In our case we set `REPLICA IDENTITY FULL` on the source postgres database so that Debezium add all columns to the deleted row and not only the id.
> 
> -> that's for the deletes
> 
> In case of the updates, the sink was sometimes timing out when comitting files to GCS and increasing the timeout fixed the issue.

gotcha, thanks for reply!

**Comment by de-donghyun-kim** [comment]
Hi @julien-alpaca ! did you solve this issues? 

I'm asking because I had similar issues. When i was updating data from Mysql Debezium kafka connector I could find duplicated data that was not deleted from iceberg table.

**Comment by julien-alpaca** [comment]
> Hi [@julien-alpaca](https://github.com/julien-alpaca) ! did you solve this issues?
> 
> I'm asking because I had similar issues. When i was updating data from Mysql Debezium kafka connector I could find duplicated data that was not deleted from iceberg table.

Yes, this can have multiple reasons. But make sure if your destination table is partitioned to pass the partition column in the delete payload.
